rfcs/convergence/lockstep-versioning.md

```diff
@@ -0,0 +1,145 @@
+
+### Changes
+
+- Updates react-accordion to use root as slot ([PR #19483](https://github.com/microsoft/fluentui/pull/19483) by bsunderhus@microsoft.com)
+```
+
+Component packages can have more concrete changelog entries, but they will still contain version bump changelog entries.
+This is problem is made worse by the fact that our daily release cadence means that any PR that day will get released,
+and all dependents are updated.
+
+After a few months, we already have >100 alpha versions of `react-components`. It is incredibly hard to for users
+to find what changed between versions. Some partners are still pinned at `alpha.60`, it will be a challenge for them
+to follow all the changelog entries to check for changes.
+
+## Detailed Design or Proposal
+
+### Rollup the react-components changelog and remove the version bump entries
+
+Work has already been done in [#19349](https://github.com/microsoft/fluentui/pull/19349), but additional work is
+required to add this kind of functionality into our CI release flow. This work **will** be completed regardless of
+the result of the other proposals in this RFC, since the `@fluentui/react-components` changelog currently provides
+no meaningful information to users.
+
+### Slower release cadence
+
+A slower release cadence will allow us to:
+
+- Plan work, PRs and validation before release.
+- Release a coherent set of changes with clear migration plan for version upgrades.
+- Reduce the number of version bump entries in changelogs that don't provide a lot of information.
+- Set clear expectations for partners when they can expect to consume expected changes/fixes. There should be very
+  little reason that partners can't plan around a slower release cadence.
+
+This RFC proposes a **monthly** automated release cadence for Fluent v9 packages. There should be a clear way to
+run ad-hoc releases when necessary, to ship urgent features and hotfixes.
+
+Daily/Nightly releases be published under a `9.x.x-nightly.commitSha` prerelease tag, for anyone that wants to
+test features from the bleeding edge.
+
+### Lockstep all v9 packages
+
+All v9 packages should be bumped in lockstep. Any release bumps every package with the maximum change
+type in the version group (major, minor, patch). Below is an example of how it would work:
+
+```
+// Before
+@fluentui/react-components 9.0.0
+@fluentui/react-button     9.0.0 -> patch
+@fluentui/react-accordion  9.0.0 -> patch
+@fluentui/react-utilities  9.0.0 -> minor
+
+
+// After
+@fluentui/react-components 9.1.0
+@fluentui/react-button     9.1.0
+@fluentui/react-accordion  9.1.0
+@fluentui/react-utilities  9.1.0
+```
+
+Once partners install `react-components` as a single suite package they can investigate versioning issues
+but checking if every v9 dependency is on the same version as the suite package. Installing separate component/utility
+packages also becomes easy, since partners should just make sure that the resolved versions are all the same.
```

Comment:
I think this is the most crucial assumption of this whole idea. End consumers need to install `react-components` to reap the benefits of this, otherwise it introduces behaviour that may not be obvious to consumers which actually complicates debugging.

As long as we are confident this will be the default approach it makes sense to me to enable lockstep versioning because you should consume all packages simply wrapped up in a single versioned package.

However, if this is not the case we can run into scenarios such as end users creating something like `react-interactibles` package containing only a subset of our interactible components, but now their package version updates as all the components updated even if we only updates something like a layout component which was not even included in their package.
I'm not saying this would be a common scenario, just showcasing how this introduces a form of tight coupling between unrelated packages.

Reply:
If the `react-interactibles` is a package that only uses a subset of our components. The only problem we would have is during a major version bump, since caret dependencies would pick up minor and patch versions.

On the other hand, if consumers used individual packages, and each had a different major bump cadence, it might turn out to be dependency hell